test/libsolidity/semanticTests/various/selfdestruct_post_cancun_same_tx.sol

```diff
@@ -0,0 +1,31 @@
+contract C {
+    event Terminated();
+
+    constructor() payable {
+        assert(address(this).balance == 1 ether);
+        assert(address(msg.sender).balance == 0);
+        emit Terminated();
+        selfdestruct(payable(msg.sender));
+    }
+}
+
+contract D {
+    C public c;
+
+    constructor() payable {
+        c = new C{value: 1 ether}();
+    }
```

Comment:
You could reduce verbosity a bit by combining this test with `selfdestruct_post_cancun.sol`. I.e. you could have `deploy()`/`terminate()` to test separate transactions and `deploy_and_terminate()` to test single transaction. Then you'd reuse the rest of the code, ending up with less boilerplate.

Then you could also reuse the same test for pre-cancun just by creating a copy with `EVMVersion: <cancun` and updating expectations.

Reply:
Yeah, good point. I will refactor these tests a bit.

Reply:
I grouped the tests expectations to reduce the number of test files.